src/database/schema/swaps_v2.ts

```diff
@@ -0,0 +1,56 @@
+import { relations } from 'drizzle-orm';
+import { pgTable, serial, varchar, integer, timestamp, boolean, unique, jsonb, numeric } from 'drizzle-orm/pg-core';
+
+import { chains } from './chains';
+import { tokens } from './tokens';
+
+// Define the swaps table without foreign key constraints on poolId
+export const swapsTableV2 = pgTable(
+  'swaps_v2',
+  {
+    id: serial('id').primaryKey(),
+    chainId: integer('chain_id')
+      .notNull()
+      .references(() => chains.id, { onDelete: 'cascade' }),
+    transactionHash: varchar('transaction_hash', { length: 256 }).notNull(),
+    user: varchar('user', { length: 256 }).notNull(),
+    baseId: varchar('base_id', { length: 256 }).notNull(),
+    quoteId: varchar('quote_id', { length: 256 }).notNull(),
+    poolId: integer('pool_id') // Removed foreign key constraint
+      .notNull(), // You can still keep it as not null if you need to ensure a value is always provided
+    dexType: varchar('dex_type', { length: 256 }).notNull(),
+    poolIdx: varchar('pool_idx', { length: 256 }).notNull(),
```

Comment:
Base and quote should be referencing to tokens table, pool - to pools table (it can be nullable in case we will have a way to make swaps without pools). 

If sql is failing to store data because of missing reference, then it means that job was ran too early (tokens and pools needs to be processed first).

poolIdx can be removed - we will have these value in the pools table.

dexType, I'm not sure we actually need too, so maybe lets just have `type` instead, which will be null by default.

```suggestion
    baseId: integer('base_id')
      .notNull()
      .references(() => tokens.id, { onDelete: 'cascade' }),
    quoteId: integer('quote_id')
      .notNull()
      .references(() => tokens.id, { onDelete: 'cascade' }),
    poolId: integer('pool_id').references(() => pools.id, { onDelete: 'cascade' }),
    type: varchar('type', { length: 64 }),
```